mob.go

```diff
@@ -527,7 +527,7 @@ func deleteRemoteWipBranch(configuration config.Configuration) {
 
 func start(configuration config.Configuration) error {
 	uncommittedChanges := hasUncommittedChanges()
-	if uncommittedChanges && !configuration.StartIncludeUncommittedChanges {
+	if uncommittedChanges && !configuration.StartIncludeUncommittedChanges && !configuration.StartDiscardUncommittedChanges {
```

Comment:
as you noticed by implementing this feature, its a flag for the user - but for us internally, its not a flag, but one of many options how to deal with uncommitted changes. So I think we should also code it this way. 

the flags should result in some sort of enum 
```
DealWithUncommittedChanges {
    Include,
    Discard,
    Ignore
}
```
where each of it leads to another path in our code.

This also eliminates the invariant where both booleans could be true (which doesnt make any sense).

The names I used in the enum are just quickly made up. You might find better names, or a better solution to implement the polymorphism, but please don't use two booleans for the same decision point.

Reply:
@gregorriegler I think you are right. It should be a enum. I adjusted it. Please have a look again :)

Reply:
i like it! feel free to merge